File: 10k app/frontend/lib/data/models/food_item.dart
class FoodItem {
  final int id;
  final String name;
  final double calories;
  final double proteinG;
  final double carbsG;
  final double fatG;
  final String? servingSize;
  final bool isCustom;
  final String? foodCategory;

  const FoodItem({
    required this.id,
    required this.name,
    required this.calories,
    required this.proteinG,
    required this.carbsG,
    required this.fatG,
    this.servingSize,
    this.isCustom = false,
    this.foodCategory,
  });

  factory FoodItem.fromJson(Map<String, dynamic> json) {
    return FoodItem(
      id: json['id'] as int,
      name: json['name'] as String,
      calories: (json['calories'] as num).toDouble(),
      proteinG: (json['proteinG'] as num).toDouble(),
      carbsG: (json['carbsG'] as num).toDouble(),
      fatG: (json['fatG'] as num).toDouble(),
      servingSize: json['servingSize'] as String?,
      isCustom: json['custom'] as bool? ?? false,
      foodCategory: json['foodCategory'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'calories': calories,
    'proteinG': proteinG,
    'carbsG': carbsG,
    'fatG': fatG,
    'servingSize': servingSize,
  };
}

File: 10k app/backend/src/main/java/com/gymflow/model/FoodItem.java
package com.gymflow.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "food_item")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class FoodItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private double calories;
    private double proteinG;
    private double carbsG;
    private double fatG;

    @Column(length = 50)
    private String servingSize;

    @Builder.Default
    private boolean isCustom = false;

    @Column(length = 30)
    private String foodCategory;
}
